components/web_discovery/renderer/blink_document_extractor.cc

```diff
@@ -0,0 +1,97 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/web_discovery/renderer/blink_document_extractor.h"
+
+#include <string>
+#include <utility>
+
+#include "third_party/blink/public/web/web_element.h"
+#include "third_party/blink/public/web/web_local_frame.h"
+
+namespace web_discovery {
+
+namespace {
+
+constexpr char kTextContentAttributeName[] = "textContent";
+
+void ProcessAttributeRequests(
+    std::string_view root_selector,
+    const std::vector<mojom::SelectAttributeRequestPtr>& requests,
+    const blink::WebVector<blink::WebElement>& elements,
+    std::vector<mojom::AttributeResultPtr>& results) {
```

Comment:
imo it would be cleaner to make this function return the vector and append every returned vector to the final container in `QueryElementAttributes`

Reply:
well that's also fair point